modules/ducktests/tests/ignitetest/tests/discovery_test.py

```diff
@@ -63,143 +75,86 @@ def __init__(self, test_context):
-        Test two-node-failure scenario (not the coordinator) with ZooKeeper.
-        """
-        self.__start_zk_quorum()
-
-        return self.__simulate_nodes_failure(version, self.__zk_properties(self.zk_quorum.connection_string()), 2)
-
-    @cluster(num_nodes=NUM_NODES+3)
-    @parametrize(version=str(DEV_BRANCH))
-    @parametrize(version=str(LATEST_2_7))
-    def test_zk_coordinator(self, version):
-        """
-        Test coordinator-failure scenario with ZooKeeper.
-        """
-        self.__start_zk_quorum()
+        if self.zk_quorum:
+            self.zk_quorum.stop()
 
-        return self.__simulate_nodes_failure(version, self.__zk_properties(self.zk_quorum.connection_string()), 0)
+    def __simulate_nodes_failure(self, version, properties, modules, config):
+        if config.nodes_to_kill == 0:
+            return {"No nodes to kill": "Nothing to do"}
 
-    def __simulate_nodes_failure(self, version, properties, nodes_to_kill=1):
-        """
-        :param nodes_to_kill: How many nodes to kill. If <1, the coordinator is the choice. Otherwise: not-coordinator
-        nodes of given number.
-        """
         self.servers = IgniteService(
             self.test_context,
-            num_nodes=self.NUM_NODES,
-            modules=["zookeeper"],
+            num_nodes=self.NUM_NODES - 1,
+            modules=modules,
             properties=properties,
             version=version)
 
-        self.stage("Starting ignite cluster")
-
         time_holder = self.monotonic()
 
         self.servers.start()
 
-        if nodes_to_kill > self.servers.num_nodes - 1:
-            raise Exception("Too many nodes to kill: " + str(nodes_to_kill))
-
         data = {'Ignite cluster start time (s)': round(self.monotonic() - time_holder, 1)}
-        self.stage("Topology is ready")
 
-        failed_nodes, survived_node = self.__choose_node_to_kill(nodes_to_kill)
+        failed_nodes, survived_node = self.__choose_node_to_kill(config.kill_coordinator, config.nodes_to_kill)
 
         ids_to_wait = [node.discovery_info().node_id for node in failed_nodes]
 
-        self.stage("Stopping " + str(len(failed_nodes)) + " nodes.")
+        if config.with_load:
+            self.__start_loading(version, properties, modules)
 
         first_terminated = self.servers.stop_nodes_async(failed_nodes, clean_shutdown=False, wait_for_stop=False)
 
-        self.stage("Waiting for failure detection of " + str(len(failed_nodes)) + " nodes.")
-
         # Keeps dates of logged node failures.
         logged_timestamps = []
 
         for failed_id in ids_to_wait:
-            self.servers.await_event_on_node(self.__failed_pattern(failed_id), survived_node, 10,
+            self.servers.await_event_on_node(self.__failed_pattern(failed_id), survived_node, 20,
                                              from_the_beginning=True, backoff_sec=0.01)
             # Save mono of last detected failure.
             time_holder = self.monotonic()
```

Comment:
how about to rename to last_blabla (as opposed to first_terminated)?

Reply:
This returns first time. Why last?

Reply:
since you have a loop, it will record last turn time.

Reply:
`first_terminated` is out the loop, it's before it. There is an atomic inside. It holds time of the first kill command.